fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1281,13 +1285,41 @@ private static void checkSearchParameterCombinations(Class<?> resourceType, List
             case TOKEN: {
                 // token
                 // [parameter]=[system]|[code]
+                // [parameter]:of-type=[system|code|value]
                 /*
                  * TODO: start enforcing this:
                  * "For token parameters on elements of type ContactPoint, uri, or boolean,
                  * the presence of the pipe symbol SHALL NOT be used - only the
                  * [parameter]=[code] form is allowed
                  */
                 String[] parts = v.split(SearchConstants.BACKSLASH_NEGATIVE_LOOKBEHIND + "\\|");
+                if (Modifier.OF_TYPE.equals(modifier)) {
+                    // Convert :of-type into a composite search parameter
+                    final String ofTypeParmName = searchParameter.getCode().getValue() + SearchConstants.OF_TYPE_MODIFIER_SUFFIX;
+                    parameterValue.setOfTypeModifier(true);
+                    if (parts.length > 1 && parts.length < 4) {
+                        QueryParameterValue typeParameterValue = new QueryParameterValue();
+                        if (parts.length == 3) {
+                            typeParameterValue.setValueSystem(unescapeSearchParm(parts[0]));
+                        }
+                        typeParameterValue.setValueCode(unescapeSearchParm(parts[parts.length - 2]));
+                        QueryParameter typeParameter = new QueryParameter(Type.TOKEN, SearchUtil.makeCompositeSubCode(ofTypeParmName, SearchConstants.OF_TYPE_MODIFIER_COMPONENT_TYPE),
+                            null, null, Collections.singletonList(typeParameterValue));
+                        parameterValue.addComponent(typeParameter);
+
+                        QueryParameterValue valueParameterValue = new QueryParameterValue();
+                        valueParameterValue.setValueCode(unescapeSearchParm(parts[parts.length - 1]));
+                        QueryParameter valueParameter = new QueryParameter(Type.TOKEN, SearchUtil.makeCompositeSubCode(ofTypeParmName, SearchConstants.OF_TYPE_MODIFIER_COMPONENT_VALUE),
+                            null, null, Collections.singletonList(valueParameterValue));
+                        parameterValue.addComponent(valueParameter);
+                    } else {
```

Comment:
Might be ok if system was not specified, but if number of parts is not 2 or 3 I think we should definitely fail it.

Reply:
Yeah, this was the one part I wasn't sure of.  Do we fail if not all these parts are provided?  It feels like allowing 2 parts would be ok, and then fail or no results if just one.  Trying to be consistent with other token searches, too.
@lmsurpre Do you have a preference?

Reply:
I don't have a strong preference on 3 parts vs. 2 parts, but I think if only 1 part then we should fail or ignore based on 'lenient' setting.

Reply:
I don't have a strong preference on this one.  I'd be fine to require either 2 or 3 parts.
I would think it would have the following semantics:
* If they use 2 parts, it means the first part is a search on the Identifier.type.code (irregardless of system)
* And if they use 3 parts, I we allow them to leave the first part empty (which should be like searching on a system-less code)

Reply:
works for me

Reply:
I'll update it to require at least 2 parts.

Reply:
> I'll update it to require at least 2 parts.
I thought it already did :-)

Reply:
If there's only 1 part, then line 1316 still produces a valid search against just the value (without any system or code).

Reply:
Updates made.  Should be ready to merge after successful build.

Reply:
I think if the :of-type modifier is specified, then anything other than 2 or 3 parts should produce an error. Think this case should be handled same as line 1300

Reply:
I can have it do that.  It will be strange that all other token search don't check for too many parts.  For example, you can search for "identifier=system|code|other1|other2|other3" and there's no error.

Reply:
I think all other token searches should generate an error too :). But for consistency's sake, I can live with it as is.

Reply:
Sounds good. I'll leave it for consistency's sake.  If we want to do stricter parsing of token searches and throw errors that can be done consistently at a later time.